src/App.tsx

```diff
@@ -1,73 +1,109 @@
-import React from 'react';
+import React, { ChangeEvent, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import cn from 'classnames';
+
+function debounce<T>(
+  callback: (args: T) => void,
+  delay: number,
+): (args: T) => void {
+  let timerId = 0;
+
+  return (...args) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdown, setIsDropdown] = useState(false);
+
+  const applyQuery = useMemo(() => debounce(setAppliedQuery, 1000), []);
+
+  const onSearchHandler = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setSelectedPerson(null);
+  };
+
+  const onSelect = (person: Person) => () => {
+    setSelectedPerson(person);
+  };
+
+  const onFocus = () => setIsDropdown(true);
+  const onBlur = () => setIsDropdown(false);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.includes(appliedQuery),
```

Comment:
why i can't to find person?
![Screenshot 2024-06-17 at 09 44 54](https://github.com/yarkob/react_autocomplete/assets/33029864/032fa9c4-f86c-4d1f-922e-3cc170507abf)
![Screenshot 2024-06-17 at 09 45 05](https://github.com/yarkob/react_autocomplete/assets/33029864/91fa0f72-f01a-4efe-b8df-ddf7bf67634e)

Reply:
do not show empty list block, when no matching suggestions